.github/workflows/tests.yml

```diff
@@ -0,0 +1,47 @@
+name: Tests
+
+on: [push]
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Cache node modules
+      uses: actions/cache@v1
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+    - name: Install dependencies (npm ci)
+      run: npm ci
+      working-directory: ./phone-store
+    - name: Run unit tests (npm run test)
+      run: npm run test -- --no-watch --no-progress --browsers=ChromeHeadlessCI
```

Comment:
Is there a reason to use `npm run test` instead of `ng test` here?

Also, would it be possible/reasonable to add the `--code-coverage` flag here and then add an expectation that the code coverage be at least X%? Or have the coverage displayed in a badge?

Reply:
I don't think there is a particular reason I'm using that command here but it is consistent with the e2e one in using `npm` rather than `ng`.

Reply:
I believe we discussed requiring code coverage for other labs and decided not to at that point but it is something we might want to explore again.

Reply:
> I don't think there is a particular reason I'm using that command here but it is consistent with the e2e one in using `npm` rather than `ng`.

I don't think it _is_ consistent anymore. The E2E tests are run with `ng e2e` and flags (which they were _not_ before), so I feel like we would want this to be `ng test` as well to minimize the number of different commands that get used.

> I believe we discussed requiring code coverage for other labs and decided not to at that point but it is something we might want to explore again.

We did, and for various reasons decided not to chase it at the time. Now would be a good time to consider re-opening that box, though. I wouldn't want that to block this PR, though, so I created a new issue for that (#4).

Reply:
You are confusing the package.json with the GitHub Actions workflow. In the workflow file (the file highlighted here) both the test and e2e commands are run through `npm run`.

Reply:
You are correct. Not sure how I got that turned around. Thanks.